opt_einsum/paths.py

```diff
@@ -652,6 +655,202 @@ def greedy(inputs, output, size_dict, memory_limit=None, choose_fn=None, cost_fn
     return ssa_to_linear(ssa_path)
 
 
+def _tree_to_sequence(c):
+    # converts a contraction tree to a contraction sequence, e.g.
+    #
+    # ((1,2),(0,(4,5,3))) --> [(1, 2), (1, 2, 3), (0, 2), (0, 1)]
+    # 
+    # 0     0         0           (1,2)       --> ((1,2),(0,(3,4,5)))
+    # 1     3         (1,2)   --> (0,(3,4,5))
+    # 2 --> 4     --> (3,4,5)
+    # 3     5
+    # 4     (1,2)
+    # 5
+    # 
+    # this function iterates through the table shown above from right to left; 
+    # note that contractions are commutative, i.e. (j,k,l) = (k,l,j);
+    # note that the solution is not unique;
+    
+    if type(c) == int:
+        return []
+    
+    c = [c] # list of remaining contractions (lower part of columns shown above)
+    t = []  # list of elementary tensors (upper part of colums)
+    s = []  # resulting contraction sequence
+    
+    while len(c) > 0:
+        j = c.pop(-1)
+        s.insert(0, tuple())
+        
+        for i in sorted([i for i in j if type(i) == int]):
+            s[0] += (sum(1 for q in t if q < i),)
+            t.insert(s[0][-1], i)
+        
+        for i in [i for i in j if type(i) != int]:
+            s[0] += (len(t) + len(c),)
+            c.append(i)
+    
+    return s
+
+
+def _sum_single_tensor_indices(inputs, output):
+    
```

Comment:
Could you document what the output etc of this is? As might be useful elsewhere!

Reply:
Sorry that I have forgotten to document this function. However, I just found a bug in my code, so maybe this function will change anyway. (Nevertheless, the final version will have a short documentation.)